regression_tests/tests/gitbook/test_docs.py

```diff
@@ -0,0 +1,181 @@
+    output = ""
+
+    try:
+        # Create a temporary file with every code block from the Markdown file.
+        with open(filename, "w") as f:
+            f.write(snippet)
+
+        # Execute the code in the file and capture the output.
+        result = subprocess.run(["python3", filename], stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True, text=True)
+
+        # Store the output from execution.
+        output = result.stdout
+    except subprocess.CalledProcessError as e:
+        success = False
+        output = e.stdout
+        error = e.stderr
+    # Delete the temporary file
+    os.remove(filename)
+    return SnippetExec(snippet, success, output, error)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--docs-folder",
+        dest="docs_folder",
+        default="gitbook",
+        action="store",
+        help="Path to gitbook directory.",
+    )
+
+    parser.add_argument(
+        "--run-subset",
+        dest="run_subset",
+        default=False,
+        action="store_true",
+        help="Run a subset of the gitbook sections.",
+    )
+
+    parser.add_argument(
+        "--sections",
+        dest="sections",
+        default=[],
+        nargs='+',
+        action="store",
+        help="Subset of gitbook sections to run.",
+    )
+
+    args = parser.parse_args()
+
+    # If we do not want to test every code snippet in the documentation, we can `run_subset` instead.
+    if args.run_subset:
+        # The subset of sections we run is specified in `sections_subset`. The for loop expects a list of tuples so we add a dummy 0.
+        args.sections = [(section, 0) for section in args.sections_subset]
+    else:
+        # Otherwise, we look at all files in all sections in the `docs_folder`.
+        args.sections = os.walk(args.docs_folder)
+
+    successfully_ran_all = True
+    for item in args.sections:
+        # Check to see if the section is excluded via `blacklist_sections`.
+        skip = False
+        for blacklist_section in blacklist_sections:
+            current_section = item[0].split("gitbook/")[-1]
+            if current_section == blacklist_section or current_section.startswith(blacklist_section+"/"):
+                skip = True
+                print(">> SKIPPING ", current_section)
+                break
+
+        # skip = False which means we want to run the files in the section.
+        if not skip:
+            for file in glob(os.path.join(item[0], '*.md')):
+                file_name = file.split("gitbook/")[-1]
+
+                # Check to see if the section is excluded via `blacklist_files`.
+                skip = False
+                for blacklist_file in blacklist_files:
+                    if file_name == blacklist_file:
+                        skip = True
+                        print(">> SKIPPING ", file_name)
+
+                # skip = False which means we want to run the file.
+                if not skip:
```

Comment:
Instead of continuing to nest the tabs, I would prefer, for both of these conditionals:
```
if skip:
     continue
<rest of code here>
```

It keeps everything a little more aligned.